src/CMakeLists.txt

```
@@ -43,6 +43,22 @@ if(NOT (BUILD_CLIENT OR BUILD_SERVER))
 endif()
 
 
+option(PRECOMPILE_HEADERS "Precompile some headers (experimental)" FALSE)
+set(PRECOMPILED_HEADERS_PATH "" CACHE FILEPATH "Path to a file listing all headers to precompile")
```

Why is this path an option?

---

Users might want to choose to put more (or less) headers into the precompiled header.

Just one example: If we change the header list in master, the user might also want to use the new list in older versions, and in branches that are based on an older version.

(The option could be removed and the user told to gitignore the file and change it in-place, but this is nicer.)

---

> gitignore the file

Do you mean people still do `git add .`? Oh well. I always use `git add -p`, to make sure I commit what I want to commit... wait, did I just hear `git commit -a`???

Anyway, if you keep the option you can set the default right here:
```suggestion
option(PRECOMPILE_HEADERS "Precompile some headers (experimental)" FALSE)
set(PRECOMPILED_HEADERS_PATH "${CMAKE_SOURCE_DIR}/src/precompiled_headers.txt" CACHE FILEPATH "Path to a file listing all headers to precompile")
```

---

> Anyway, if you keep the option you can set the default right here: [...]

I'm not doing this, because it would make it harder to reset it to the default. I.e. `cmake . -DPRECOMPILED_HEADERS_PATH=""` would fail.